CharlieBackend.Business/Services/HomeworkStudentService.cs

```diff
@@ -0,0 +1,199 @@
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+        private readonly ILogger<HomeworkStudentService> _logger;
+
+        public HomeworkStudentService(IUnitOfWork unitOfWork, IMapper mapper, ILogger<HomeworkStudentService> logger)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+            _logger = logger;
+        }
+
+        public async Task<Result<HomeworkStudentDto>> CreateHomeworkFromStudentAsync(HomeworkStudentRequestDto homeworkStudent, ClaimsPrincipal userContext)
+        {
+            long accountId = Convert.ToInt32(userContext.Claims.First(x => x.Type.EndsWith("AccountId")).Value);
+
+            var student = await _unitOfWork.StudentRepository.GetStudentByAccountIdAsync(accountId);
+            var homework = await _unitOfWork.HomeworkRepository.GetByIdAsync(homeworkStudent.HomeworkId);
+
+            var errors = await ValidateHomeworStudentRequest(homeworkStudent, student, homework).ToListAsync();
+            
+            if (errors.Any())
+            {
+                var errorsList = string.Join("; ", errors);
+
+                _logger.LogError("Homework create request has failed due to: " + errorsList);
+
+                return Result<HomeworkStudentDto>.GetError(ErrorCode.ValidationError, errorsList);
+            }
+
+            var newHomework = new HomeworkStudent
+            {
+                StudentId = student.Id,
+                HomeworkId = homework.Id,
+                Homework = homework,
+                HomeworkText = homeworkStudent.HomeworkText
+            };
+
+            _unitOfWork.HomeworkStudentRepository.Add(newHomework);
+
+            if (homeworkStudent.AttachmentIds?.Count > 0)
+            {
+                var attachments = await _unitOfWork.AttachmentRepository.GetAttachmentsByIdsAsync(homeworkStudent.AttachmentIds);
+
+                newHomework.AttachmentOfHomeworkStudents = new List<AttachmentOfHomeworkStudent>();
+
+                foreach (var attachment in attachments)
+                {
+                    newHomework.AttachmentOfHomeworkStudents.Add(new AttachmentOfHomeworkStudent
+                    {
+                        AttachmentId = attachment.Id,
+                        Attachment = attachment,
+                    });
+                }
+            }
+
+            await _unitOfWork.CommitAsync();
+
+            _logger.LogInformation($"Homework with id {newHomework.Id} for student with id {accountId} has been added ");
+
+            return Result<HomeworkStudentDto>.GetSuccess(_mapper.Map<HomeworkStudentDto>(newHomework));
+        }
+
+        public async Task<Result<HomeworkStudentDto>> UpdateHomeworkFromStudentAsync(HomeworkStudentRequestDto homeworkStudent, ClaimsPrincipal userContext, long homeworkId)
+        {
+            long accountId = Convert.ToInt32(userContext.Claims.First(x => x.Type.EndsWith("AccountId")).Value);
```

Comment:
Please, update using `ICurrentUserService`

Reply:
I have done this